### PR TITLE
Compile using thrift 0.12.0

### PR DESCRIPTION
**Pull request by amrhassan**
Hello,

I've recompiled the `parquet.thrift` file using thrift 0.12.0 to be able to use the more idiomatically-named enums.

What do you think the new package version should be? I'm not sure if it would be useful to stick only to the [parquet-format](https://github.com/apache/parquet-format) version since the thrift compiler is also being iterated on and their release schedules should not coincide. How about a completely separate versioning scheme for this package? That would mean a major version bump for this PR since it breaks public API.

### CODE REVIEW

**Comment by sadikovi** [comment]
Makes sense to me. Would be good to release the latest format and mark it that is built with thrift 0.12.0. Is there a way to force compiler not to change struct and enum names?

**Comment by amrhassan** [comment]
>  Would be good to release the latest format and mark it that is built with thrift 0.12.0.

Yes, I've bumped the major version to indicate the API change. I don't think we should encode both the thrift compiler and parquet-format versions in the package version since[ Cargo assumes versions to follow 3-number semantic versioning](https://doc.rust-lang.org/cargo/reference/manifest.html#the-version-field).

> Is there a way to force compiler not to change struct and enum names?

Not that I can tell. I think this is going to be the default for the thrift compiler from 0.12.0 onwards, since this is the idiomatic Rust naming scheme.

**Comment by amrhassan** [comment]
Hey, any chance we can get this merged and get a new release of the crate soon?

File: Cargo.toml
Comment (sadikovi) [inline Cargo.toml]: Why is this version 3.0.0? We keep the version in sync with parquet-format version.

**Review by sadikovi — COMMENTED** [review]
Is this change backward compatible? If we push a newer version of the crate, is it going to break anything?

**Comment by sadikovi** [comment]
ping @sunchao, I left a few comments, can you take a look? Thanks!

File: src/parquet_format.rs
Comment (sunchao) [inline src/parquet_format.rs:39]: Just curious: is there a way to avoid these changes with Thrift 0.12.0? It will be great if we can avoid a backward incompatible change caused by these.

File: Cargo.toml
Comment (sunchao) [inline Cargo.toml]: Can we use semantic versions like `ordered-float = 1.0`? we can automatically get updates from minor version changes.

File: Cargo.toml
Comment (sunchao) [reply Cargo.toml]: I think this is because of the backward incompatible changes from Thrift 0.12.0. I think we should stick to 2.6.X if we can avoid that. Otherwise, we may have to bump up the major version.

**Review by sunchao — COMMENTED** [review]
Thanks @sadikovi for pinging! just left some comments.

File: Cargo.toml
Comment (amrhassan) [reply Cargo.toml]: Indeed it's because of the backwards incompatible changes introduced by Thrift 0.12.0. According to [the changelog](https://github.com/apache/thrift/tree/master/lib/rs#thrift-0130) this seems to be non-optional.

File: Cargo.toml
Comment (amrhassan) [reply Cargo.toml]: Yes sure.

File: src/parquet_format.rs
Comment (amrhassan) [reply src/parquet_format.rs:39]: It doesn't seem like it unfortunately.

**Review by sunchao — APPROVED** [review]
LGTM. I think we have to upgrade the Thrift version at some point and this should be easily fixable by search-and-replace. We should update the README as well since the version no longer matches that of Parquet version.

@sadikovi let me know if you are OK with this too.

**Comment by sadikovi** [comment]
I disagree that we should use 3.0.0 version - it is confusing. How would I know what parquet format version it is? 

I propose that the we at least include a thrift version, e.g. "2.6.0-thrift12" or something similar. If not, then we should add a section of release notes in README and say that we break compatibility for the format starting with this commit and newest release of 2.6.0.

**Comment by amrhassan** [comment]
Version `2.6.0-thrift12` works for me, but it doesn't indicate that there's breaking change in the new crate version since the major version wasn't bumped. I would say that's unexpected for every other programmer who is expecting libraries to abide by semantic versioning.

**Comment by sadikovi** [comment]
@sunchao Let's just update the code and publish crate for 2.6.0. However, I am proposing we keep crate version same as parquet-format version.

**Comment by amrhassan** [comment]
I've reverted the change to the crate version.

**Comment by sunchao** [comment]
Thanks @amrhassan and @sadikovi : to confirm, this PR also updated the Parquet thrift version to 2.6.0?

**Comment by amrhassan** [comment]
Yes, it was updated in `master` since 9748cd6dfa6faf0cd5f6678a3b59380ae2065d4a but the crate was never published I guess.

**Comment by sunchao** [comment]
Cool. @sadikovi : can you take a final look at this and approve it too? after that I'll merge it. Thanks.

**Review by sadikovi — APPROVED** [review]
LGTM.

**Comment by sunchao** [comment]
Merged. Thanks @amrhassan !!

**Comment by amrhassan** [comment]
Thank you. Any chance you could publish the crate for `2.6.0` soon?

**Comment by sunchao** [comment]
Sure. Just published the new version.

**Pull request merged**
